App crashes when rotating screen in PlaylistActivity

`08-28 13:07:05.454 6735-6735/com.mart.listlock E/AndroidRuntime: FATAL EXCEPTION: main
                                                                 Process: com.mart.listlock, PID: 6735
                                                                 java.lang.RuntimeException: Unable to pause activity {com.mart.listlock/com.mart.listlock.playactivity.PlayActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.util.List com.mart.listlock.playactivity.MusicService.getSongs()' on a null object reference
                                                                     at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2518)
                                                                     at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4077)
                                                                     at android.app.ActivityThread.-wrap15(ActivityThread.java)
                                                                     at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1350)
                                                                     at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                     at android.os.Looper.loop(Looper.java:148)
                                                                     at android.app.ActivityThread.main(ActivityThread.java:5417)
                                                                     at java.lang.reflect.Method.invoke(Native Method)
                                                                     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
                                                                     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
                                                                  Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.util.List com.mart.listlock.playactivity.MusicService.getSongs()' on a null object reference
                                                                     at com.mart.listlock.playactivity.PlayActivity.onPause(PlayActivity.java:381)
                                                                     at android.app.Activity.performPause(Activity.java:6348)
                                                                     at android.app.Instrumentation.callActivityOnPause(Instrumentation.java:1311)
                                                                     at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2496)
                                                                     at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4077) 
                                                                     at android.app.ActivityThread.-wrap15(ActivityThread.java) 
                                                                     at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1350) 
                                                                     at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                                     at android.os.Looper.loop(Looper.java:148) 
                                                                     at android.app.ActivityThread.main(ActivityThread.java:5417) 
                                                                     at java.lang.reflect.Method.invoke(Native Method) 
                                                                     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726) 
                                                                     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616) 
`

--- a/app/src/main/java/com/mart/listlock/playactivity/PlayActivity.java
+++ b/app/src/main/java/com/mart/listlock/playactivity/PlayActivity.java
@@ -373,27 +373,29 @@ public class PlayActivity extends AppCompatActivity {
         super.onPause();
         LogW.d(LOG_TAG, "paused");
 
-        SharedPreferences settings = getSharedPreferences(getString(R.string.app_name), 0);
-        SharedPreferences.Editor editor = settings.edit();
+        if (musicBound) {
+            SharedPreferences settings = getSharedPreferences(getString(R.string.app_name), 0);
+            SharedPreferences.Editor editor = settings.edit();
 
-        final List<SpotifySong> songs = musicService.getSongs();
+            final List<SpotifySong> songs = musicService.getSongs();
 
-        final int size = songs.size();
+            final int size = songs.size();
 
-        editor.putInt(KEY_SIZE, size);
+            editor.putInt(KEY_SIZE, size);
 
-        for (int i = 0; i < size; i++) {
-            final String uri = songs.get(i).getURI();
-            final boolean locked = songs.get(i).isLocked();
-            editor.putString(KEY_SONG + i, uri);
-            editor.putBoolean(KEY_SONG_LOCKED + i, locked);
-            LogW.d(LOG_TAG, "saving song URI: " + uri);
-            LogW.d(LOG_TAG, "saving song locked: " + locked);
+            for (int i = 0; i < size; i++) {
+                final String uri = songs.get(i).getURI();
+                final boolean locked = songs.get(i).isLocked();
+                editor.putString(KEY_SONG + i, uri);
+                editor.putBoolean(KEY_SONG_LOCKED + i, locked);
+                LogW.d(LOG_TAG, "saving song URI: " + uri);
+                LogW.d(LOG_TAG, "saving song locked: " + locked);
+            }
+
+            editor.apply();
+
+            unregisterReceiver(playbackEventReceiver);
         }
-
-        editor.apply();
-
-        unregisterReceiver(playbackEventReceiver);
     }
 
     @Override

